internal/testing/e2e/query_backfills_test.go

```diff
@@ -0,0 +1,200 @@
+	require.NoError(t, err)
+
+	resp, err := stream.Recv()
+	require.Equal(t, io.EOF, err)
+	require.Nil(t, resp)
+}
+
+func TestQueryBackfillsPaging(t *testing.T) {
+	om := newOM(t)
+
+	pageSize := 10 // TODO: read from config
+	if pageSize < 1 {
+		require.Fail(t, "invalid page size")
+	}
+
+	total := pageSize*5 + 1
+	expectedIds := map[string]struct{}{}
+
+	for i := 0; i < total; i++ {
+		resp, err := om.Frontend().CreateBackfill(context.Background(), &pb.CreateBackfillRequest{Backfill: &pb.Backfill{}})
+		require.NotNil(t, resp)
+		require.NoError(t, err)
+
+		expectedIds[resp.Id] = struct{}{}
+	}
+
+	stream, err := om.Query().QueryBackfills(context.Background(), &pb.QueryBackfillsRequest{Pool: &pb.Pool{}})
+	require.NoError(t, err)
+
+	foundIds := map[string]struct{}{}
+
+	for i := 0; i < 5; i++ {
+		var resp *pb.QueryBackfillsResponse
+		resp, err = stream.Recv()
+		require.NoError(t, err)
+		require.Equal(t, pageSize, len(resp.Backfills))
+
+		for _, backfill := range resp.Backfills {
+			foundIds[backfill.Id] = struct{}{}
+		}
+	}
+
+	resp, err := stream.Recv()
+	require.NoError(t, err)
+	require.Equal(t, len(resp.Backfills), 1)
+	foundIds[resp.Backfills[0].Id] = struct{}{}
+
+	require.Equal(t, expectedIds, foundIds)
+
+	resp, err = stream.Recv()
+	require.Equal(t, err, io.EOF)
+	require.Nil(t, resp)
+}
+
+func TestBackfillQueryAfterMMFUpdate(t *testing.T) {
+	ctx := context.Background()
+	om := newOM(t)
+	backfill := &pb.Backfill{
+		SearchFields: &pb.SearchFields{
+			StringArgs: map[string]string{
+				"field": "value",
+			},
+		},
+	}
+	pool := &pb.Pool{
+		StringEqualsFilters: []*pb.StringEqualsFilter{{StringArg: "field", Value: "value"}},
+	}
+	match := &pb.Match{
+		MatchId: "1",
+		Tickets: []*pb.Ticket{},
+	}
+	{
+		resp, err := om.Frontend().CreateBackfill(ctx, &pb.CreateBackfillRequest{Backfill: backfill})
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		match.Backfill = resp
+	}
+
+	om.SetMMF(func(ctx context.Context, profile *pb.MatchProfile, out chan<- *pb.Match) error {
+		out <- match
+		return nil
+	})
+
+	om.SetEvaluator(func(ctx context.Context, in <-chan *pb.Match, out chan<- string) error {
+		p, ok := <-in
+		require.True(t, ok)
+
+		out <- p.MatchId
+		return nil
+	})
+	{
+		stream, err := om.Backend().FetchMatches(ctx, &pb.FetchMatchesRequest{
+			Config:  om.MMFConfigGRPC(),
+			Profile: &pb.MatchProfile{},
+		})
+		require.Nil(t, err)
+
+		resp, err := stream.Recv()
+		require.Nil(t, err)
+		require.True(t, proto.Equal(match, resp.Match))
+
+		resp, err = stream.Recv()
+		require.Nil(t, resp)
+		require.Equal(t, io.EOF, err)
+	}
+	{
+		stream, err := om.Query().QueryBackfills(context.Background(), &pb.QueryBackfillsRequest{Pool: pool})
+		require.NoError(t, err)
+		resp, err := stream.Recv()
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.Equal(t, int64(firstBackfillGeneration), resp.Backfills[0].Generation)
```

Comment:
I think it's worth checking require.NotEmpty(t, resp.Backfills)

Reply:
Or `require.Len(t, resp.Backfills, 1)` before this line.